src/drive/mobile/modules/authorization/DriveMobileRouter.jsx

```diff
@@ -30,27 +30,27 @@ class DriveMobileRouter extends Component {
       isAppBooted: true
     })
   }
-  afterAuthentication = async () => {
-    const { client } = this.props
+
+  initClientAndBar = async client => {
+    const { saveServerUrl, saveCredentials, setToken } = this.props
     const accesstoken = client.getStackClient().token
     restoreCozyClientJs(
       client.getStackClient().uri,
       client.getStackClient().oauthOptions,
       client.getStackClient().token
     )
+    setToken(accesstoken)
     await initBar(client)
-    this.props.saveServerUrl(client.getStackClient().uri)
+    saveServerUrl(client.getStackClient().uri)
     setCozyUrl(client.getStackClient().uri)
-    this.props.saveCredentials(client, accesstoken)
+    saveCredentials(client, accesstoken)
```

Comment:
`saveCredentials ` also calls `setToken `, right? Maybe this could use a comment or a little more cleaning up?